ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/SessionProxy.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    @Override
+    public void close() throws JMSException {
+        if (transactionHelper.isTransactionAvailable()) {
+            transactionHelper.delistResource(xaSession.getXAResource());
+
+            if (LOGGER.isLoggable(Level.FINE)) {
+                LOGGER.fine("Delisted " + xaSession
+                        + " XA resource from the transaction");
+            }
+
+            Synchronization synchronization =
+                    new SessionClosingSynchronization(xaSession);
+            transactionHelper.registerSynchronization(synchronization);
+
+            if (LOGGER.isLoggable(Level.FINE)) {
+                LOGGER.fine("Registered synchronization to close the session: "
+                        + synchronization);
+            }
+
+        } else {
+            xaSession.close();
+        }
+    }
+
+    @Override
+    public BytesMessage createBytesMessage() throws JMSException {
+        return xaSession.createBytesMessage();
+    }
+
+    @Override
+    public MapMessage createMapMessage() throws JMSException {
+        return xaSession.createMapMessage();
+    }
+
+    @Override
+    public Message createMessage() throws JMSException {
+        return xaSession.createMessage();
+    }
+
+    @Override
+    public ObjectMessage createObjectMessage() throws JMSException {
+        return xaSession.createObjectMessage();
+    }
+
+    @Override
+    public ObjectMessage createObjectMessage(Serializable serializable)
+            throws JMSException {
+
+        return xaSession.createObjectMessage(serializable);
+    }
+
+    @Override
+    public StreamMessage createStreamMessage() throws JMSException {
+        return xaSession.createStreamMessage();
+    }
+
+    @Override
+    public TextMessage createTextMessage() throws JMSException {
+        return xaSession.createTextMessage();
+    }
+
+    @Override
+    public TextMessage createTextMessage(String s) throws JMSException {
+        return xaSession.createTextMessage(s);
+    }
+
+    @Override
+    public boolean getTransacted() throws JMSException {
+        return xaSession.getTransacted();
+    }
+
+    @Override
+    public int getAcknowledgeMode() throws JMSException {
+        return xaSession.getAcknowledgeMode();
+    }
+
+    @Override
+    public void commit() throws JMSException {
+        // TODO maybe throw exception straight away since this should not happen?
+        xaSession.commit();
+    }
+
+    @Override
+    public void rollback() throws JMSException {
+        // TODO maybe throw exception straight away since this should not happen?
+        xaSession.rollback();
+    }
+
+    @Override
+    public void recover() throws JMSException {
+        xaSession.recover();
+    }
+
+    @Override
+    public MessageListener getMessageListener() throws JMSException {
+        return xaSession.getMessageListener();
+    }
+
+    @Override
+    public void setMessageListener(MessageListener messageListener)
+            throws JMSException {
+
```

Comment:
What's wrong with the empty line after the wrapped method declaration? We have that in many places. (At least in all my code)